### PR TITLE
Cleanup the repo resource

### PR DESCRIPTION
**Pull request by olive42**
Make it similar to resources written more recently.

### CODE REVIEW

File: resources/repo.rb [hunk 1]
```diff
@@ -1,28 +1,19 @@
...
+property :repo_name, String, name_attribute: true
+property :repo_type, String, default: 'maven2-hosted'
 property :attributes, Hash, default: lazy { ::Mash.new } # Not mandatory but strongly recommended in the generic case.
 property :online, [true, false], default: true
-property :api_endpoint, String, desired_state: false, identity: true,
-                                default: lazy { node['nexus3']['api']['endpoint'] }
-property :api_username, String, desired_state: false, identity: true,
-                                default: lazy { node['nexus3']['api']['username'] }
-property :api_password, String, desired_state: false, identity: true, sensitive: true,
+property :api_endpoint, String, desired_state: false, default: lazy { node['nexus3']['api']['endpoint'] }
```
Comment (Annih) [inline resources/repo.rb]: should be `identity: true` otherwise you don't have access to it in `load_current_value`

File: resources/repo.rb [hunk 2]
```diff
@@ -1,28 +1,19 @@
...
+property :api_username, String, desired_state: false, default: lazy { node['nexus3']['api']['username'] }
```
Comment (Annih) [inline resources/repo.rb]: should be `identity: true` otherwise you don't have access to it in `load_current_value`

File: resources/repo.rb [hunk 3]
```diff
@@ -1,28 +1,19 @@
...
+property :api_password, String, desired_state: false, sensitive: true,
```
Comment (Annih) [inline resources/repo.rb]: should be `identity: true` otherwise you don't have access to it in `load_current_value`

File: resources/repo.rb [hunk 4]
```diff
@@ -1,28 +1,19 @@
...
-  api_password desired.api_password
-
-  def apiclient
-    @apiclient ||= ::Nexus3::Api.new(api_endpoint, api_username, api_password)
-  end
-
   begin
-    res = apiclient.run_script('get_repo', desired.repo_name)
+    res = ::Nexus3::Api.new(api_endpoint, api_username, api_password).run_script('get_repo', desired.repo_name)
     current_value_does_not_exist! if res == 'null'
```
Comment (Annih) [inline resources/repo.rb]: Handle this case with `config.nil?` after `JSON.parse`

File: resources/repo.rb (hunk 1 above)
Comment (olive42) [reply resources/repo.rb]: done

File: resources/repo.rb (hunk 2 above)
Comment (olive42) [reply resources/repo.rb]: done

File: resources/repo.rb (hunk 3 above)
Comment (olive42) [reply resources/repo.rb]: done

File: resources/repo.rb (hunk 4 above)
Comment (olive42) [reply resources/repo.rb]: done

**Pull request merged**